src/components/user/auth/signupModal.js

```diff
@@ -0,0 +1,301 @@
+                          {this.state.phoneError}
+                        </h6>
+                      )}
+                    </Form.Group>
+                    <Form.Group controlId="signupFormBasicTextArea">
+                      <Form.Label>Short Description</Form.Label>
+                      <Form.Control
+                        onChange={this.onFieldChange}
+                        as="textarea"
+                        rows={3}
+                        name="userShortDescription"
+                      />
+                      {this.state.userShortDescriptionError && (
+                        <h6 className="form-field-error">
+                          {this.state.userShortDescriptionError}
+                        </h6>
+                      )}
+                    </Form.Group>
+                    <Row>
+                      <Col md={6}>
+                        <Form.Group controlId="signupFormBasicPassword1">
+                          <Form.Label>Password</Form.Label>
+                          <Form.Control
+                            onChange={this.onFieldChange}
+                            type="password"
+                            name="password"
+                          />
+                          {this.state.passwordError && (
+                            <h6 className="form-field-error">
+                              {this.state.passwordError}
+                            </h6>
+                          )}
+                        </Form.Group>
+                      </Col>
+                      <Col md={6}>
+                        <Form.Group controlId="signupFormBasicPassword2">
+                          <Form.Label>Re-Type Password</Form.Label>
+                          <Form.Control
+                            onChange={this.onFieldChange}
+                            type="password"
+                            name="verifyPassword"
+                          />
+                          {this.state.verifyPasswordError && (
+                            <h6 className="form-field-error">
+                              {this.state.verifyPasswordError}
+                            </h6>
+                          )}
+                        </Form.Group>
+                      </Col>
+                    </Row>
+                    <Button
+                      className="primary-button"
+                      variant=""
+                      type="submit"
+                      disabled={this.state.isFormInvalid}
+                    >
+                      Sign Up
+                    </Button>
+                  </Form>
+                </div>
+              </Col>
+            </Row>
+          </Container>
+        </Modal.Body>
+      </Modal>
+    );
+  }
+}
+
+const mapStateToProps = (state) => {
+  return {
+    isLoggedIn: state.auth.isLoggedIn,
+    error: state.auth.error,
+  };
+};
+
+const mapDispatchToProps = (dispatch) => {
+  return {
```

Comment:
if possible create actions for this in separate files and then just use functions in the components to dispatch the actions

Reply:
for references check actions in donut frontend

Reply:
Afaik actions are only used when something happens or to trigger something which might change store. Here I am calling the dispatch function itself; so any benefit of creating a separate action, then making a function to dispatch that action which will eventually end up in calling this dispatch function only. For reference, I am using [createSlice](https://redux.js.org/tutorials/essentials/part-2-app-structure#creating-slice-reducers-and-actions) which automatically creates respective actions for the defined reducers.

Reply:
Actions are used to dispatch the request to reducers to let them know what to change and how to change, action itself doesn't change the store, it is handled by reducers.

Not sure if it is good to use createSlice, seen the first time in the code.

Reply:
createSlice is a newer addition as a combination of createAction and createReducer. With this, we don't explicitly have to write action creators but just a bunch of reducers and initial state (and the respective required actions get created automatically).
So, to change/update stuff in store, we sort of dispatch the required reducer function call; which creates a new action object and dispatches it to the reducers eventually updating the store.
Imo this won't be a problem; just a different way to write same code